### PR TITLE
Add missing doc files and cleanup nits

### PR DESCRIPTION
**Pull request by dougbu**
- nit: Remove useless `$(HasReferenceAssembly)` settings …
- nit: Remove useless `$(CompileUsingReferenceAssemblies)` settings …
- Add more `$(GenerateDocumentationFile)` settings
- nit: Remove useless `$(IsPackable)` settings
- nit: Remove useless `$(IsProjectReferenceProvider)` settings
- nit: Remove useless `$(IsTestAssetProject)` settings

- best to review this PR commit-by-commit
    - 4815b2d4c1e2 is the most interesting change: Adds 5 missing doc files
          - please confirm these files are useful @pranavkm @javiercn

### CODE REVIEW

File: src/Identity/ApiAuthorization.IdentityServer/src/Microsoft.AspNetCore.ApiAuthorization.IdentityServer.csproj [hunk 1]
```diff
@@ -5,9 +5,6 @@
     <Description>ASP.NET Core API Authorization package powered by Identity Server.</Description>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
     <PackageTags>aspnetcore;apiauth;identity</PackageTags>
-    <HasReferenceAssembly>false</HasReferenceAssembly>
-    <IsPackable>true</IsPackable>
-    <HasReferenceAssembly>false</HasReferenceAssembly>
```
Comment (pranavkm) [inline src/Identity/ApiAuthorization.IdentityServer/src/Microsoft.AspNetCore.ApiAuthorization.IdentityServer.csproj:10]: Making doubly sure!

**Review by pranavkm — APPROVED** [review]

**Comment by pranavkm** [comment]
You might have broken packaging for helix tests

File: src/Identity/ApiAuthorization.IdentityServer/src/Microsoft.AspNetCore.ApiAuthorization.IdentityServer.csproj (hunk 1 above)
Comment (javiercn) [reply src/Identity/ApiAuthorization.IdentityServer/src/Microsoft.AspNetCore.ApiAuthorization.IdentityServer.csproj:10]: This is not part of the shared framework. Doesn't need a ref assembly.

Comment (dougbu) [reply src/Identity/ApiAuthorization.IdentityServer/src/Microsoft.AspNetCore.ApiAuthorization.IdentityServer.csproj:10]: It won't get one. The default is `false` for any project that doesn't set `$(IsAspNetCoreApp)` to `true`.

Comment (javiercn) [reply src/Identity/ApiAuthorization.IdentityServer/src/Microsoft.AspNetCore.ApiAuthorization.IdentityServer.csproj:10]: Wasn't aware of that.

**Comment by BrennanConroy** [comment]
Can you double check the packages in shipping/non-shipping. The windows packages artifacts are ~76MB smaller in this PR.

**Review by JunTaoLuo — APPROVED** [review]
Spot checking the changes seems okay. But obviously the CI disagrees. Also, any comments on @BrennanConroy's feedback?

**Comment by dougbu** [comment]
I'm rebuilding locally to see what's going on.

**Comment by dougbu** [comment]
> Can you double check the packages in shipping/non-shipping. The windows packages artifacts are ~76MB smaller in this PR.

I did this in exhaustive detail over the last couple of days, while hunting for a root cause of the IIS NewShim test failures. Didn't find anything except the expected new doc files in a couple of packages. There is some random variation of symbol sizes (a few bytes here or there) but nothing else when comparing builds of a specific 'master' commit against these changes on top of the same commit.

**Comment by BrennanConroy** [comment]
> I did this in exhaustive detail over the last couple of days

Umm, it took 5 seconds to find the difference. This PR doesn't produce "AspNetCoreRuntime.5.0.x64" and "AspNetCoreRuntime.5.0.x86" packages in Shipping. I don't know what those are used for or if they're important or not.

**Comment by dougbu** [comment]
Those are site extensions packages and they're produced only in official builds. Were correctly created in my local all-encompassing builds.

**Pull request merged**